lightning/src/ln/channelmanager.rs

```diff
@@ -787,7 +826,8 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 		let channel = Channel::new_outbound(&self.fee_estimator, &self.keys_manager, their_network_key, channel_value_satoshis, push_msat, user_id, config)?;
 		let res = channel.get_open_channel(self.genesis_hash.clone());
 
-		let _consistency_lock = self.total_consistency_lock.read().unwrap();
+		let _scopeguard = ReadLockNotifierGuard::new(&self.total_consistency_lock, &self.persistence_notifier);
```

Comment:
Perhaps `_scopeguard` is a little redundant given that "guard" is used for a lock lasting the duration of a scope. Depending on the outcome of my earlier comment, perhaps `_persistence_guard`.